spec/hosts/example_spec.rb

```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+describe 'testhost.example.com' do
+
+  let(:facts) {{ :osfamily => 'RedHat', :concat_basedir  => '/dne', :define_fact => "" }}
+
+  context 'When given connected records that depend on each other' do
+    it { should compile }
```

Comment:
This test isn't testing anything yet because testhost.example.com is not yet defined.

Can you copy in the file from
https://github.com/ajjahn/puppet-dns/pull/80/files#diff-4f9b2b877cece59826df5b65a4980421R1

Adding that file and ensuring it can compile will be a definitive test that this all integrates correctly and the dependency bug is fixed, and will stay fixed.